elf_ctx.c

```diff
@@ -166,12 +199,51 @@ void elf_ctx_init(struct ukarch_ctx *ctx, struct elf_prog *prog,
 		{ AT_PHNUM, prog->phdr.num },
 		{ AT_PHDR, (__uptr)prog->vabase + prog->phdr.off },
 #if CONFIG_APPELFLOADER_VDSO
+		/* TODO: This must also be pushed and copied
+		 * or mapped to information block. Move it to infoblk_auxv
+		 * ASAP.
+		 */
 		{ AT_SYSINFO_EHDR, (uintptr_t)vdso_image_addr },
 #endif /* CONFIG_APPELFLOADER_VDSO */
 		{ AT_IGNORE, 0x0 }
 	};
 	struct auxv_entry auxv_null = { AT_NULL, 0x0 };
 
+	for (i = (int)ARRAY_SIZE(infoblk_auxv) - 1; i >= 0; i--) {
```

Comment:
I wanted to keep it as `int` since I want to be able to reach index `0` and I want the loop to naturally stop once we reach a negative index. To avoid the warning of assigning with different signedness I had to cast.